gcloud/bigquery/dataset.py

```diff
@@ -190,6 +190,25 @@ def location(self, value):
             raise ValueError("Pass a string, or None")
         self._properties['location'] = value
 
+    @classmethod
+    def from_api_repr(cls, resource, client):
+        """Factory:  construct a dataset given its API representation
+
+        :type resource: dict
+        :param resource: dataset resource representation returned from the API
+
+        :type client: :class:`gcloud.bigquery.client.Client`
+        :param client: Client which holds credentials and project
+                       configuration for the dataset.
+
+        :rtype: :class:`gcloud.bigquery.dataset.Dataset`
+        :returns: Dataset parsed from ``resource``.
+        """
+        name = resource['datasetReference']['datasetId']
```

Comment:
These exist to parse server-generate resources:  the `datasetReference` structure is *always* there (unlike some of the other properties).  Maybe we should rename them to `_from_api_repr` to indicate that they are private.